R/detect_bad_recursion.R

```diff
@@ -0,0 +1,33 @@
+#' Detect bad recursion in function call
+#'
+#' Examines the stack trace that is passed as an argument to determine
+#' if import::*() has been called recursively in a way that is not
+#' allowed (i.e. bad).
+#'
+#' Any recursive use of import::*() within a module that is itself being
+#' imported should only be done with import::here(). Never with
+#' import::from() or import::into()
+#'
+#' A bad recursion is a recursion in which the
+#'
+#' @param stack_trace A list object containing the value of a \code{.traceback()} call
+#' @return logical indicating whether a bad recursion call was detected
+#'
+#' @noRd
+detect_bad_recursion <- function(stack_trace) {
```

Comment:
Will the [redirects/dispatch to `from`](https://github.com/smbache/import/blob/master/R/here.R#L18) cause problems?

Reply:
I think not, that is what lines 27-29 are intended to do. If we find ourselves in "good recursion" (the inner call was made through `import::here()`, then the trace (after grepping only for lines starting with `import`:

```
[1] "import::from(.from = module_recursive_inner.R, to_title, .into = {"
[2] "import::here(module_recursive_inner.R, to_title)"                  
[3] "import::from(module_recursive_outer_here.R, print_title_text)"  
```

Basically, an arbitrary number of  import statements, but the last two (the only ones we examine at this point, because any outer statements were presumably either legal or have been fixed already) should be first a `from()` (because that's the only place we perform the test, and then a `here()` (where the dispatch originated). 

Not claiming this is elegant, but logically it seems to me like it should work, and the tests also seem to bear that out.

Reply:
✅